addons/escoria-core/game/core-scripts/esc/esc_event_manager.gd

```diff
@@ -124,7 +127,13 @@ func _process(delta: float) -> void:
 			if event_flags & ESCEvent.FLAG_NO_SAVE:
 				escoria.save_manager.save_enabled = false
 
-			_running_events[channel_name].run()
+			var rc = _running_events[channel_name].run()
```

Comment:
Won't this be an issue for events happening while the player has control? For example, activate an alarm so that the guard turns back to the player for a few seconds, during while the player has to perform a specific action and remain unseen?

Reply:
As noted in the Discord channel: Yeah, I hadn't considered the other channels. I'll try implementing a "per channel yield" check and we can test that out.

Reply:
This has been put in. I'll mark this as resolved but please do test.